src/formats/apple-notes/convert-note.ts

```diff
@@ -0,0 +1,385 @@
+	}
+	
+	formatAttr(attr: ANAttributeRun): string {
+		switch (attr.fontWeight) {
+			case ANFontWeight.Bold:
+				attr.fragment = `**${attr.fragment}**`;
+				break;
+			case ANFontWeight.Italic:
+				attr.fragment = `*${attr.fragment}*`;
+				break;
+			case ANFontWeight.BoldItalic:
+				attr.fragment = `***${attr.fragment}***`;
+				break;
+		}
+		
+		if (attr.strikethrough) attr.fragment = `~~${attr.fragment}~~`;
+		if (attr.link && attr.link != attr.fragment) attr.fragment = `[${attr.fragment}](${attr.link})`;
+		
+		if (attr.atLineStart) return this.formatParagraph(attr);
+		else return attr.fragment;
+	}
+	
+	formatParagraph(attr: ANAttributeRun): string {
+		const indent = '\t'.repeat(attr.paragraphStyle?.indentAmount || 0);
+		const styleType = attr?.paragraphStyle?.styleType;
+		let prelude = attr.paragraphStyle?.blockquote ? '> ' : '';
+		
+		if (
+			this.listNumber != 0 &&
+			(styleType !== ANStyleType.NumberedList ||
+			this.listIndent !== attr.paragraphStyle?.indentAmount)
+		) {
+			this.listIndent = attr.paragraphStyle?.indentAmount || 0;
+			this.listNumber = 0;
+		}
+		
+		switch (styleType) {
+			case ANStyleType.Title:
+				return `${prelude}# ${attr.fragment}`;
+				
+			case ANStyleType.Heading:
+				return `${prelude}## ${attr.fragment}`;
+				
+			case ANStyleType.Subheading:
+				return `${prelude}### ${attr.fragment}`;
+				
+			case ANStyleType.DashedList:
+			case ANStyleType.DottedList:
+				return `${prelude}${indent}- ${attr.fragment}`;
+				
+			case ANStyleType.NumberedList:
+				this.listNumber++;
+				return `${prelude}${indent}${this.listNumber}. ${attr.fragment}`;
+				
+			case ANStyleType.Checkbox:
+				const box = attr.paragraphStyle!.checklist?.done ? '[x]' : '[ ]';
+				return `${prelude}${indent}- ${box} ${attr.fragment}`;
+		}
+		
+		//Not a list but indented in line with one
+		if (this.multiRun == ANMultiRun.List) prelude += indent;
+		
+		return `${prelude}${attr.fragment}`;
+	}
+	
+	async formatAttachment(attr: ANAttributeRun): Promise<string> {
+		let row, id;
+		
+		switch (attr.attachmentInfo.typeUti) {
+			case ANAttachment.Hashtag:
+			case ANAttachment.Mention:
+				row = await this.importer.database.get`
+					SELECT zalttext FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				return row.ZALTTEXT;
+				
+			case ANAttachment.InternalLink:
+				row = await this.importer.database.get`
+					SELECT ztokencontentidentifier FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				const destIdentifier = row.ZTOKENCONTENTIDENTIFIER.match(NOTE_URI)[1];
+				
+				row = await this.importer.database.get`
+					SELECT z_pk FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${destIdentifier.toUpperCase()}`;
+					
+				if (!(row.Z_PK in this.importer.resolvedFiles)) {
+					//if we don't have the note yet, we fetch it early to get its path
+					this.importer.resolveNote(row.Z_PK);
```

Comment:
Does this work as expected if two notes reference each other forming a cycle?

Reply:
Yes, it has been designed to avoid that (per my earlier comments, the note is intentionally added to `resolvedFiles` before its content is parsed)